### PR TITLE
e2e: only select linux nodes for NodeLogQuery tests

### PR DESCRIPTION
**Pull request by aravindhp**
#### What type of PR is this?

/kind bug
/kind failing-test

#### What this PR does / why we need it:

The current NodeLogQuery tests only work against Linux nodes, so explicitly select them.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/node/kubelet.go [hunk 1]
```diff
@@ -460,12 +462,18 @@ var _ = SIGDescribe("kubelet", func() {
...
 		ginkgo.BeforeEach(func(ctx context.Context) {
 			nodes, err := e2enode.GetReadyNodesIncludingTainted(ctx, c)
 			framework.ExpectNoError(err)
 			if len(nodes.Items) == 0 {
-				framework.Fail("Expected at least one Linux node to be present")
+				framework.Fail("Expected at least one node to be present")
 			}
+			linuxNodes := getLinuxNodes(nodes)
+			if len(linuxNodes.Items) == 0 {
+				framework.Fail("Expected at least one node to be present")
```
Comment (jsturtevant) [inline test/e2e/node/kubelet.go:474]:
```suggestion
				framework.Fail("Expected at least one Linux node to be present")
```

File: test/e2e/node/kubelet.go [hunk 2]
```diff
@@ -460,12 +462,18 @@ var _ = SIGDescribe("kubelet", func() {
 	f.Describe("kubectl get --raw \"/api/v1/nodes/<insert-node-name-here>/proxy/logs/?query=/<insert-log-file-name-here>", feature.NodeLogQuery, "[LinuxOnly]", func() {
 		var linuxNodeName string
 
+		framework.Logf("AZURE_SSH_KEY: %s", os.Getenv("AZURE_SSH_KEY"))
```
[the remark this answers is not in the document]
Comment (aravindhp) [reply test/e2e/node/kubelet.go]: 🤦🏽‍♂️ Fixed

File: test/e2e/node/kubelet.go (hunk 1 above)
Comment (aravindhp) [reply test/e2e/node/kubelet.go]: Done.

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Pull request merged**